src/main/java/io/confluent/connect/elasticsearch/ElasticsearchSinkConnectorConfig.java

```diff
@@ -152,6 +155,23 @@
       + "Elasticsearch rejects due to some malformation of the document itself, such as an index"
       + " mapping conflict or a field name containing illegal characters. Valid options are "
       + "'ignore', 'warn', and 'fail'.";
+  public static final String WRITE_METHOD_CONFIG = "write.method";
+  private static final String WRITE_METHOD_DOC = "What method to use for storing data to "
+          + "Elasticsearch. We can use one of " + WriteMethod.INSERT.toString() + ", "
+          + WriteMethod.UPDATE.toString() + " or " + WriteMethod.UPSERT.toString()
+          + ". In case of " +  WriteMethod.INSERT.toString() + ", documents are inserted. "
+          + "If document with same ID already exists it is fully replaced. If "
+          + WriteMethod.UPDATE.toString() + " is used, documents will be updated. I.e. "
+          + "only fields present in payload will be updated (or added if original document "
+          + "does not contain them). Other fields stays untouched. However, if document with "
+          + "specified ID does not exist, update will fail. " + WriteMethod.UPSERT.toString()
+          + " works similar way " + WriteMethod.UPDATE.toString() + " does, but if document "
+          + "with specified ID does not exists it will be created. Valid options are "
+          + Arrays.toString(WriteMethod.names()).toLowerCase(Locale.ROOT)
+          + ". Default to '" + WriteMethod.DEFAULT.toString() + "'.\n"
+          + "If Elasticsearch has limited resources, " + WriteMethod.UPSERT.toString()
+          + " may take significant amount of time. You may want to modify properties "
+          + FLUSH_TIMEOUT_MS_CONFIG + ", " + READ_TIMEOUT_MS_CONFIG + " and " + BATCH_SIZE_CONFIG;
```

Comment:
As a user, I'd want to know whether this affected the existing delete behavior (no), and whether the default of `insert` matched previous behavior (yes). And this could be improved to follow the style of our documentation.

For example, maybe:
> Method used for writing data to Elasticsearch, and one of `insert`, `update` or `upsert`. The default method is `insert`, in which the connector constructs a document from the record value and inserts that document into Elasticsearch, completely replacing any existing document with the same ID; this matches previous behavior. The `upsert` method will create a new document if one with the specified ID does not yet exist, or will update an existing document with the same ID by adding/replacing only those fields present in the record value. The `update` method will update an existing document with the same ID by adding/replacing only those fields present in the record value, but will fail if the document with the specified ID does not exist. The `upsert` method may require additional time and resources of Elasticsearch, so consider increasing the `flush.timeout.ms`, `read.timeout.ms`, and `batch.size` configuration properties. 

However, I'm struggling to understand the value of `update` and envision a realistic use case where `write.method=update` might be useful. My concern is that even if an edge use case does exist, users with more conventional use cases might struggle by accidentally setting it to `update` instead of `upsert`. Unless I'm missing something, perhaps it's better to start without `update` yet keep all the other code as-is so that we can add it later if it does become useful.

Reply:
OK, I will replace current doc text with one you suggest.
I will keep the code as-is but valid values for `write.method` will be reduced to `insert` and `upsert`.